django/contrib/postgres/forms/array.py

```diff
@@ -130,11 +132,10 @@ def render(self, name, value, attrs=None, renderer=None):
                 widget_value = None
             if id_:
                 final_attrs = dict(final_attrs, id='%s_%s' % (id_, i))
-            output.append(self.widget.render(name + '_%s' % i, widget_value, final_attrs, renderer))
-        return mark_safe(self.format_output(output))
-
-    def format_output(self, rendered_widgets):
-        return ''.join(rendered_widgets)
+            context['subwidgets'].append(
+                self.widget.render(name + '_%s' % i, widget_value, final_attrs)
```

Comment:
This PR looks good. It would be slightly more consistent with `SelectDate` and `Multiwidget` if this render was handled in the template. The `SelectDate` widget does something similar where the widget type is instantiated for each subfield,  `get_context` is called, and the `widget` return value is added to `subwidgets`:

https://github.com/django/django/blob/3e91850dccecd13dde8cef7b81c798217f74a301/django/forms/widgets.py#L961